src/audio/eq_iir/eq_iir.c

```diff
@@ -704,250 +734,158 @@ static int eq_iir_verify_params(struct comp_dev *dev,
+ */
+static void eq_iir_set_alignment(struct audio_stream *source, struct audio_stream *sink)
+{
+	const uint32_t byte_align = 1;
+	const uint32_t frame_align_req = 1;
 
-	return 0;
+	audio_stream_init_alignment_constants(byte_align, frame_align_req, source);
+	audio_stream_init_alignment_constants(byte_align, frame_align_req, sink);
 }
 
-static int eq_iir_prepare(struct comp_dev *dev)
+static int eq_iir_prepare(struct processing_module *mod)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd = module_get_private_data(mod);
+	struct module_data *md = &mod->priv;
 	struct comp_buffer *sourceb, *sinkb;
 	struct comp_buffer __sparse_cache *source_c, *sink_c;
+	struct comp_dev *dev = mod->dev;
 	enum sof_ipc_frame source_format;
 	enum sof_ipc_frame sink_format;
-	uint32_t sink_period_bytes;
+	uint32_t sink_period_bytes, source_period_bytes;
 	int ret;
 
-	comp_info(dev, "eq_iir_prepare()");
-
-	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
+	ret = eq_iir_verify_params(dev, mod->stream_params);
 	if (ret < 0)
 		return ret;
 
-	if (ret == COMP_STATUS_STATE_ALREADY_SET)
-		return PPL_STATUS_PATH_STOP;
+	comp_info(dev, "eq_iir_prepare()");
 
 	/* EQ component will only ever have 1 source and 1 sink buffer */
-	sourceb = list_first_item(&dev->bsource_list,
-				  struct comp_buffer, sink_list);
-	sinkb = list_first_item(&dev->bsink_list,
-				struct comp_buffer, source_list);
+	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
+	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
 
 	source_c = buffer_acquire(sourceb);
 	sink_c = buffer_acquire(sinkb);
 
-	/* get source data format */
-	source_format = source_c->stream.frame_fmt;
+	eq_iir_set_alignment(&source_c->stream, &sink_c->stream);
 
-	/* get sink data format and period bytes */
+	/* get source and sink data format */
+	source_format = source_c->stream.frame_fmt;
 	sink_format = sink_c->stream.frame_fmt;
-	sink_period_bytes = audio_stream_period_bytes(&sink_c->stream,
-						      dev->frames);
 
+	source_period_bytes = audio_stream_period_bytes(&source_c->stream, dev->frames);
+	sink_period_bytes = audio_stream_period_bytes(&sink_c->stream, dev->frames);
 	if (sink_c->stream.size < sink_period_bytes) {
 		comp_err(dev, "eq_iir_prepare(): sink buffer size %d is insufficient < %d",
 			 sink_c->stream.size, sink_period_bytes);
-		ret = -ENOMEM;
-		goto out;
+		buffer_release(sink_c);
+		buffer_release(source_c);
+		return -ENOMEM;
 	}
 
+	md->mpd.in_buff_size = source_period_bytes;
+	md->mpd.out_buff_size = sink_period_bytes;
+
 	cd->config = comp_get_data_blob(cd->model_handler, NULL, NULL);
 
 	/* Initialize EQ */
 	comp_info(dev, "eq_iir_prepare(), source_format=%d, sink_format=%d",
 		  source_format, sink_format);
+
 	if (cd->config) {
-		ret = eq_iir_setup(cd, source_c->stream.channels);
+		ret = eq_iir_setup(mod, source_c->stream.channels);
+		buffer_release(sink_c);
+		buffer_release(source_c);
```

Comment:
you now have 3 locations where you release buffers. Wouldn't it be better to keep the original `if {} else {}` code structure and release the buffers once at the end?